unicef/etools
# Feature flag v2

Connects unicef/etools-issues#599

This moves the 'feature flag' functionality to the `environment` app, and to the develop branch. I spent quite a bit of time trying to build a UI for it, but my attempts didn't seem to be better than the Django admin UI that waffle (plus my minor tenant-specific addtion) provides, so before doing that, I think I'll need some feedback on exactly how this is going to be used, by whom, and what the UI needs to look like. I'd prefer for that to be a separate PR on top of this, since this is functional as-is.

We use the [django-waffle](https://github.com/jsocol/django-waffle) package and I created a new app called `features` to modify it slightly to allow segmentation of users by 'country'.

To test this PR:
1. go to the bottom of the admin page and under Waffle, click 'Add Flag'
2. Create a flag named 'can_view_foo' and assign it a few countries (way at the bottom of the form). Make sure that you uncheck the superuser box otherwise the flag will always be on.
3. Go back to the dashboard (/dash/) and make sure you're in the workspace of a country that you added
to the flag.
4. Load /api/v2/environment/flags/ -> should return JSON `{active_flags: ['can_view_foo']}`
5. Go back to the dashboard and choose a different country
6. Load /api/v2/environment/flags/ -> should return JSON `{active_flags: []}`

Please let me know if this looks like a good plan and then I'll add tests and docs, and will start fleshing out the requirements for a UI, if we feel a UI beyond the admin is needed.
